dashboard/app/models/user.rb

```diff
@@ -1308,4 +1308,25 @@ def show_school_info_interstitial?
   def school_info_suggestion?
     !(school.blank? && full_address.blank?)
   end
+
+  # Removes PII and other information from the user and marks the user as having been purged.
+  # WARNING: This (permanently) destroys data and cannot be undone.
+  # WARNING: This does not purge the user, only marks them as such.
+  def clear_user_and_mark_purged
+    self.name = nil
+    self.username = 'abcdef'
```

Comment:
Ah, that was meant to be a placeholder... Asking offline, will update thread with resolution (conversation if it is enlightening).